src/app/common/hocs/withLists.jsx

```diff
@@ -0,0 +1,92 @@
+import { Component } from 'react'
+import isEmpty from 'lodash/isEmpty'
+import get from 'lodash/get'
+import memizedOne from 'memoize-one'
+import Loading from '../widgets/Loading'
+
+function filterData(data = [], filter = '') {
+  if(isEmpty(data)) { return [] }
+  if(!filter) { return data }
+  return data.filter(({ name }) => name.toLowerCase().includes(filter.toLowerCase()))
+}
+
+const cachedFiler = memizedOne(filterData)
+
+export default function ListsHOC(WrappedComponent) {
+  return class withListData extends Component {
+    componentDidMount() {
+      this.getInitialData()
+    }
+
+    static defaulProps = {
+      apiParams: {},
+      filters: {},
+      showLoading: true,
+    }
+
+    loadNext = () => {
+      if(isEmpty(this.props.data) || this.props.isLoading) { return }
+      const { count, results } = this.props.data || {}
+      if(this.props.usePages || this.props.useLimit) {
+        if(results.length >= count || this.props.isLoading) {
+          return
+        }
+        const { page, offset, limit } = this.props.filters || {}
+        if(this.props.usePages) {
+          return this.props.fetch({ ...this.props.filters, page: page + 1, ...this.props.apiParams })
+        }
+        if(this.props.useLimit) {
+          return this.props.fetch({
+            ...this.props.filters, offset: offset + limit, limit: 25, ...this.props.apiParams,
```

Comment:
`limit` used very often. I see that we can rewrite it at any time, but maybe will be better to set this value somewhere as constant?
Let's imagine that you would want to set it one time with value 30 and use this value in all project.

Reply:
agree, default limit can be in env